docker/Dockerfile.builder

```diff
@@ -1,27 +1,39 @@
+# syntax=docker/dockerfile:1.7-labs
+
 # Target with dependencies to build all flow tools from their sources.
 # i.e., "./build_openroad.sh --local" from inside a docker container
 # NOTE: don't use this file directly unless you know what you are doing,
 # instead use etc/DockerHelper.sh
+
 ARG fromImage=openroad/flow-ubuntu22.04-dev:latest
 
-FROM $fromImage AS openroad-builder-base
+FROM $fromImage AS orfs-base
 
-ARG numThreads=$(nproc)
-
-COPY . /OpenROAD-flow-scripts
 WORKDIR /OpenROAD-flow-scripts
+COPY --link dev_env.sh dev_env.sh
+COPY --link build_openroad.sh build_openroad.sh
 
-RUN ./build_openroad.sh --no_init --local --threads ${numThreads}
-
-FROM $fromImage AS openroad-flow-scripts-base
+FROM orfs-base AS orfs-builder-base
 
-COPY . /OpenROAD-flow-scripts
-
-RUN rm -rf /OpenROAD-flow-scripts/tools /OpenROAD-flow-scripts/.git
-
-COPY --from=openroad-builder-base /OpenROAD-flow-scripts/tools/install /OpenROAD-flow-scripts/tools/install
-
-FROM $fromImage
+COPY --link tools tools
+ARG numThreads=$(nproc)
 
-COPY --from=openroad-flow-scripts-base /OpenROAD-flow-scripts /OpenROAD-flow-scripts
-WORKDIR /OpenROAD-flow-scripts
+RUN echo "" > tools/yosys/abc/.gitcommit && \
+  ./build_openroad.sh --no_init --local --threads ${numThreads}
+
+FROM orfs-base
+
+COPY --link flow/platforms flow/platforms
+COPY --link flow/designs flow/designs
+COPY --link flow/tutorials flow/tutorials
```

Comment:
These are still not sorted... Can chatgpt write a small script? :-)

Reply:
As you suggested in a previous comment, those layers are sorted by `update size` (lines)  in descending order.
Let me know if there are any other modifications to the order you want to see.

Here's a look into the statistics for those layers (calculated on this branch starting from the initial commit):
| Name            | Commits | Updated lines    | Docker layer size |
|-----------------|---------|------------|------------|
| flow/platforms  | 519     | 53224454   |  729 MB |
| flow/designs    | 1763    | 11026554   | 284 MB |
| flow/tutorials  | 10      | 36408      | 4.4 MB |
| docs            | 108     | 7146       |  4.1 MB |
| flow/util       | 702     | 18579      |  240 KB |
| flow/scripts    | 839     | 16455      | 112 KB |
| flow/test       | 74      | 291        | 8 KB
| docker          | 9       | 75         | 2 KB

Reply:
You also need to take update frequency into account...

So flow/designs change MUCH more often than tutorials for instance and show be lower down.

Reply:
@ivysochyn 
Could you share a GH Gitst of the script command line you used to generate the data? This looks interesting and useful for gathering data from other repos.

Reply:
I suspect the rebuilding the <10mb layers is trivial in runtime and designs + platforms is most of the time.  If true, the table order is reasonable.

I expect frequency is most-to-least: scripts, designs, platforms, util, ...

Reply:
Taking into account both the frequency of commits and the size of the layers, I would rather prioritize frequency. This results in the following order:

| Name            | Commits | Updated lines | Docker layer size |
|-----------------|---------|---------------|-------------------|
| docker          | 9       | 75            | 2 KB              |
| flow/tutorials  | 10      | 36408         | 4.4 MB            |
| docs            | 108     | 7146          | 4.1 MB            |
| flow/test       | 74      | 291           | 8 KB              |
| flow/platforms  | 519     | 53224454      | 729 MB            |
| flow/util       | 702     | 18579         | 240 KB            |
| flow/scripts    | 839     | 16455         | 112 KB            |
| flow/designs    | 1763    | 11026554      | 284 MB            |

This order aims to optimize the cache usage, as the first layers are less likely to be rebuilt frequently.

Additionally, in the changes introduced in this PR, I rely on the [COPY --link](https://docs.docker.com/reference/dockerfile/#copy---link) behavior, which should treat all of those layers independently, preserving the cache even if previous layers are invalidated.

For your reference, here are the tools and commands I used to collect this data.

To calculate the number of commits:
```
git log --oneline docs | wc -l
```

Lines updated calculation:
```
git log --shortstat -- <FILE/DIRECTORY> | grep "files changed" | awk '{inserts += $4; deletes += $6} END {print "Insertions: " inserts "\nDeletions: " deletes "\nTotal Changes: " inserts + deletes}
```

To see details of each specific layer of the image, I usually go with [dive](https://github.com/wagoodman/dive):
```
docker run -ti --rm  -v /var/run/docker.sock:/var/run/docker.sock wagoodman/dive <DOCKER_IMAGE_TAG>
```

Reply:
Makes sense to.me!